napari/layers/_shapes_layer/model.py

```diff
@@ -227,6 +217,12 @@ def __init__(
                 z_index=z_index,
             )
 
+            if len(self._data) > 1:
+                # If multiple ShapeLists present and 2D slice key still there
+                # from initialization then remove 2D slice key
```

Comment:
I don't understand this comment at all, could you elaborate?

Reply:
i fixed this, had to do with how things were initialized, but i've made it much better now i think